oak-run/src/test/java/org/apache/jackrabbit/oak/index/DocumentStoreIndexerIT.java

```diff
@@ -380,6 +387,95 @@ protected CompositeIndexer prepareIndexers(NodeStore nodeStore, NodeBuilder buil
 
     }
 
+    @Test
+    public void metrics() throws Exception {
+        MongoConnection mongoConnection = getConnection();
+        DocumentNodeStoreBuilder<?> docBuilder = builderProvider.newBuilder()
+                .setMongoDB(mongoConnection.getMongoClient(), mongoConnection.getDBName());
+        DocumentNodeStore store = docBuilder.build();
+
+        Whiteboard wb = new DefaultWhiteboard();
+        MongoDocumentStore ds = (MongoDocumentStore) docBuilder.getDocumentStore();
+        Registration r1 = wb.register(MongoDocumentStore.class, ds, emptyMap());
+
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        MetricStatisticsProvider metricsStatisticsProvider = new MetricStatisticsProvider(null, executor);
+        wb.register(StatisticsProvider.class, metricsStatisticsProvider, emptyMap());
+        Registration c1Registration = wb.register(MongoDatabase.class, mongoConnection.getDatabase(), emptyMap());
+
+        configureIndex(store);
+
+        NodeBuilder builder = store.getRoot().builder();
+        NodeBuilder appNB = newNode("app:Asset");
+        createChild(appNB,
+                "jcr:content",
+                "jcr:content/comments",
+                "jcr:content/metadata",
+                "jcr:content/metadata/xmp",
+                "jcr:content/renditions",
+                "jcr:content/renditions/original",
+                "jcr:content/renditions/original/jcr:content"
+        );
+        builder.child("test").setChildNode("book.jpg", appNB.getNodeState());
+        store.merge(builder, EmptyHook.INSTANCE, CommitInfo.EMPTY);
+
+        String checkpoint = store.checkpoint(100000);
+
+        //Shut down this store and restart in readOnly mode
+        store.dispose();
```

Comment:
Why is it necessary to call `dispose()` twice? Perhaps the test could be divided into 2?

Reply:
This variable is used to reference two different objects. And I agree that the test could be improved, but I just copy'n'pasted the existing test and do not want to spend much more time in refactoring it.